src/RelationalBehavior.php

```diff
@@ -113,40 +126,71 @@ public function saveRelations($event)
     {
         /** @var ActiveRecord $model */
         $model = $event->sender;
+        $position = 0;
 
         $relatedRecords = $model->getRelatedRecords();
         foreach ($relatedRecords as $relationName => $relationRecords) {
             $activeQuery = $model->getRelation($relationName);
             if (!empty($activeQuery->via)) { // works only for many-to-many relation
                 list($junctionTable, $primaryModelColumn, $relatedModelColumn) = $this->parseQuery($activeQuery);
-
                 $junctionRows = [];
                 $relationPks = ArrayHelper::getColumn($relationRecords, array_keys($activeQuery->link)[0], false);
                 $passedRecords = count($relationPks);
-                $relationPks = array_filter($relationPks);
+
+                if ($this->sortSettings[$relationName] &&
+                    isset($this->sortSettings[$relationName]['sortColumn']) &&
+                    isset($this->relationalRawData[$relationName])
+                ) {
+                    $relationPks = $this->relationalRawData[$relationName];
```

Comment:
my logic is pretty strightfoward:
I get $relationPks on [124 line](https://github.com/E96/yii2-relational-behavior/pull/1/files/3b07f0e47594f4e34722c572d7f60f5e4d64c9a7#diff-beaf5a48bea9d3765c0ecae2f84aadddL124)
Then I filter them on [126 line](https://github.com/E96/yii2-relational-behavior/pull/1/files/3b07f0e47594f4e34722c572d7f60f5e4d64c9a7#diff-beaf5a48bea9d3765c0ecae2f84aadddL126)

In your code, you overwrite existing $relationPks at [144 line](https://github.com/E96/yii2-relational-behavior/pull/1/files/3b07f0e47594f4e34722c572d7f60f5e4d64c9a7#diff-beaf5a48bea9d3765c0ecae2f84aadddR144). This is not good.